ncompare/core.py

```diff
@@ -193,59 +226,48 @@ def compare_two_nc_files(out: Outputter,
                          ) -> tuple[int, int, int]:
     """Go through all groups and all variables, and show them side by side - whether they align and where they don't."""
     out.side_by_side(' ', 'File A', 'File B')
-
     num_var_diffs = {"left": 0, "right": 0, "both": 0}
-    with netCDF4.Dataset(nc_one) as nc_a, netCDF4.Dataset(nc_two) as nc_b:
 
+    with netCDF4.Dataset(nc_one) as nc_a, netCDF4.Dataset(nc_two) as nc_b:
         out.side_by_side('All Variables', ' ', ' ', dash_line=False)
         out.side_by_side('-', '-', '-', dash_line=True)
-        out.side_by_side('num variables in root group:', len(nc_a.variables), len(nc_b.variables))
-        out.side_by_side('-', '-', '-', dash_line=True)
-
         # Count differences between the lists of variables in the root group.
-        num_var_diffs['left'], num_var_diffs['right'], num_var_diffs['both'] = \
-            count_diffs(nc_a.variables, nc_b.variables)
+        # Utilizing set operations for efficiency
+        vars_a_set = set(nc_a.variables)
+        vars_b_set = set(nc_b.variables)
+        common_vars = vars_a_set & vars_b_set
+        num_var_diffs['both'] = len(common_vars)
+        num_var_diffs['left'] = len(vars_a_set - common_vars)
+        num_var_diffs['right'] = len(vars_b_set - common_vars)
+
+        out.side_by_side('num variables in root group:', len(vars_a_set), len(vars_b_set))
+        out.side_by_side('-', '-', '-', dash_line=True)
 
         # Go through root-level variables.
-        for variable_pair in common_elements(nc_a.variables, nc_b.variables):
+        for variable in common_vars:
```

Comment:
@danielfromearth The pleasure is all mine. This is a great tool and I was very impressed to see that you are maintaining it all by yourself. I love speeding up Python programs in my free time and this one looked challenging.

If you could post a known set of reference files I can test against, along with the exact command you use for the reference output, it would be really appreciated and helpful.

With that said, I shall also wait for your changes in the test suite and update my PR accordingly.